### PR TITLE
feat: use new $createdAt and $updatedAt attributes

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

Using the newly introduced `$createdAt` and `$updatedAt` attributes and replacing similar old ones on existing collections.

Following Response Models now have those attributes:
- Bucket
- Collection
- Deployment
- Document
- Domain
- Execution
- File
- Func
- Key
- Membership
- Platform
- Project
- Session (only `$createdAt`)
- Team
- Token (only `$createdAt`)
- User
- Webhook

Following attributes have been removed from existing Collections:
- Collections
- Platforms
- Functions
- Teams
- Deployments
- Executions
- Buckets
- Files

## Test Plan

Existing tests.

## Related PRs and Issues

#3011 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: app/config/collections.php [hunk 1]
```diff
@@ -2639,17 +2520,6 @@
         '$id' => 'files',
         '$name' => 'Files',
         'attributes' => [
-            [
```
Comment (Meldiron) [inline app/config/collections.php:2642]: Regardidng removing existing ones, please also look at:

- https://github.com/appwrite/appwrite/blob/0dd6c25b49204b02b83dc1d4a894147ca0a01c05/app/config/collections.php#L669-L679
- https://github.com/appwrite/appwrite/blob/0dd6c25b49204b02b83dc1d4a894147ca0a01c05/app/config/collections.php#L995-L1005
- https://github.com/appwrite/appwrite/blob/0dd6c25b49204b02b83dc1d4a894147ca0a01c05/app/config/collections.php#L1540-L1549
-

File: app/init.php [hunk 2]
```diff
@@ -461,6 +461,11 @@ function (mixed $value) {
         ->withPassword($dbPass)
         ->withOptions([
             PDO::ATTR_ERRMODE => App::isDevelopment() ? PDO::ERRMODE_WARNING : PDO::ERRMODE_SILENT, // If in production mode, warnings are not displayed
+            PDO::ATTR_TIMEOUT => 3, // Seconds
+            PDO::ATTR_PERSISTENT => true,
+            PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
+            PDO::ATTR_EMULATE_PREPARES => true,
+            PDO::ATTR_STRINGIFY_FETCHES => true,
```
Comment (Meldiron) [inline app/init.php:468]: Is this on purpose, or development leftovers?

File: app/init.php [hunk 3]
```diff
@@ -553,11 +558,12 @@ function (mixed $value) {
     $dbScheme = App::getEnv('_APP_DB_SCHEMA', '');
 
     $pdo = new PDO("mysql:host={$dbHost};port={$dbPort};dbname={$dbScheme};charset=utf8mb4", $dbUser, $dbPass, array(
-        PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8mb4',
         PDO::ATTR_TIMEOUT => 3, // Seconds
         PDO::ATTR_PERSISTENT => true,
         PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
         PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
+        PDO::ATTR_EMULATE_PREPARES => true,
+        PDO::ATTR_STRINGIFY_FETCHES => true,
```
Comment (Meldiron) [inline app/init.php:566]: Is this on purpose, or development leftovers?

File: src/Appwrite/Utopia/Response/Model/Session.php [hunk 4]
```diff
@@ -16,6 +16,12 @@ public function __construct()
                 'default' => '',
                 'example' => '5e5ea5c16897e',
             ])
+            ->addRule('$createdAt', [
```
Comment (Meldiron) [inline src/Appwrite/Utopia/Response/Model/Session.php:19]: rule for updatedAt missing. On purpose?

File: src/Appwrite/Utopia/Response/Model/Token.php [hunk 5]
```diff
@@ -16,6 +16,12 @@ public function __construct()
                 'default' => '',
                 'example' => 'bb8ea5c16897e',
             ])
+            ->addRule('$createdAt', [
```
Comment (Meldiron) [inline src/Appwrite/Utopia/Response/Model/Token.php:19]: updateAt missing. On purpose?

File: tests/e2e/Services/Database/DatabaseCustomServerTest.php [hunk 6]
```diff
@@ -821,7 +821,7 @@ public function testIndexLimitException()
         $this->assertIsArray($collection['body']['attributes']);
         $this->assertIsArray($collection['body']['indexes']);
         $this->assertCount(64, $collection['body']['attributes']);
-        $this->assertCount(61, $collection['body']['indexes']);
+        $this->assertCount(59, $collection['body']['indexes']);
```
Comment (Meldiron) [inline tests/e2e/Services/Database/DatabaseCustomServerTest.php:824]: Let's improve our existing database tests. There we never had createAt/updateAt. Let's make tests where we check createdAt is created properly, and updateAt updated when updating document.

Comment (Meldiron) [reply tests/e2e/Services/Database/DatabaseCustomServerTest.php:824]: Talking about that, lets also add tests to ensure people cannot touch createdAt and updatedAt. So we can present this as server-managed (secure) info.

File: app/views/console/database/collection.phtml [hunk 7]
```diff
@@ -568,8 +568,8 @@ $logs = $this->getParam('logs', null);
 
                     <ul class="margin-bottom-large text-fade text-size-small">
                         <li class="margin-bottom-small"><i class="icon-angle-circled-right margin-start-tiny margin-end-tiny"></i> <button data-ls-ui-trigger="open-json" class="link text-size-small">View as JSON</button></li>
-                        <li class="margin-bottom-small"><i class="icon-angle-circled-right margin-start-tiny margin-end-tiny"></i> Last Updated: <span data-ls-bind="{{project-collection.dateUpdated|dateText}}"></span></li>
-                        <li class="margin-bottom-small"><i class="icon-angle-circled-right margin-start-tiny margin-end-tiny"></i> Created: <span data-ls-bind="{{project-collection.dateCreated|dateText}}"></span></li>
+                        <li class="margin-bottom-small"><i class="icon-angle-circled-right margin-start-tiny margin-end-tiny"></i> Last Updated: <span data-ls-bind="{{project-collection.$updatedAt|dateText}}"></span></li>
```
Comment (Meldiron) [inline app/views/console/database/collection.phtml:571]: Suggestion: Show updatedAt and createdAt just like we show other `$` attributes:

CleanShot 2022-06-20 at 13 46 27

File: app/init.php (hunk 3 above)
Comment (TorstenDittmann) [reply app/init.php:566]: on purpose, this is coming for the new database version :)

File: src/Appwrite/Utopia/Response/Model/Session.php (hunk 4 above)
Comment (TorstenDittmann) [reply src/Appwrite/Utopia/Response/Model/Session.php:19]: sessions cannot be updated

File: src/Appwrite/Utopia/Response/Model/Token.php (hunk 5 above)
Comment (TorstenDittmann) [reply src/Appwrite/Utopia/Response/Model/Token.php:19]: tokens cannot be updated

File: app/config/collections.php (hunk 1 above)
Comment (TorstenDittmann) [reply app/config/collections.php:2642]: the `$updatedAt` or `$createdAt` are modified automatically on create or update. Sometimes we rely on a differen value.

For example with domains

File: app/views/console/database/collection.phtml (hunk 7 above)
Comment (Meldiron) [reply app/views/console/database/collection.phtml:571]: Let's show the date then. Maybe a different design to better indicate that timestamp would be copied? I think we have to show it, at least in document view if nowhere else. I would maybe even say it's valuable in the list of documents.

**Review by Meldiron — APPROVED** [review]
LGTM

**Pull request merged**